lib/install/opsuser/opsuser.go

```diff
@@ -184,9 +202,24 @@ func (mgr *RBACManager) setupPermissions(ctx context.Context) ([]rbac.ResourcePe
 		resourceDescs = append(resourceDescs, resourceDesc{rbac.Network, *netRef})
 	}
 
-	// Add resource pools
-	for _, rPoolRef := range mgr.configSpec.ComputeResources {
-		resourceDescs = append(resourceDescs, resourceDesc{rbac.Endpoint, rPoolRef})
+	// In a DRS environment, add the endpoint role to the resource pool(s). In a non-DRS
+	// environment, add the endpoint role to the cluster.
+	if mgr.session.DRSEnabled != nil && *mgr.session.DRSEnabled {
+		for _, rPoolRef := range mgr.configSpec.ComputeResources {
+			resourceDescs = append(resourceDescs, resourceDesc{rbac.Endpoint, rPoolRef})
+		}
+	} else {
+		resourceDescs = append(resourceDescs, resourceDesc{rbac.Endpoint, cluster.Reference()})
```

Comment:
@hickeng For non-DRS envs, we've had to up-level where the endpoint role is applied here - from the resource pool to the cluster. I've tested this in a non-DRS env with a VCH and containers using volumes and container networks.

Does applying the endpoint role to the cluster in a non-DRS env satisfy minimum privilege?

Reply:
We were planning on assigning the permissions to the folder and not the cluster - did that fall through for some reason? I'm unclear why they need to be assigned at a cluster level for VC without DRS if they are also assigned to the VCH folder.

Reply:
Opened https://github.com/vmware/vic/issues/7606 for follow-on work per our conversation.